storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogMetadataSerdeTest.java

```diff
@@ -0,0 +1,113 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class RemoteLogMetadataSerdeTest {
+
+    public static final String TOPIC = "foo";
+    private static final TopicIdPartition TP0 = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition(TOPIC, 0));
+    private final Time time = new MockTime(1);
+
+    @Test
+    public void testRemoteLogSegmentMetadataSerde() {
+        RemoteLogSegmentMetadata remoteLogSegmentMetadata = createRemoteLogSegmentMetadata();
+
+        doTestRemoteLogMetadataSerde(remoteLogSegmentMetadata);
+    }
+
+    @Test
+    public void testRemoteLogSegmentMetadataUpdateSerde() {
+        // Create RemoteLogSegmentMetadataUpdate for RemoteLogSegmentMetadataUpdate
+        RemoteLogSegmentMetadataUpdate remoteLogSegmentMetadataUpdate = createRemoteLogSegmentMetadataUpdate();
+
+        doTestRemoteLogMetadataSerde(remoteLogSegmentMetadataUpdate);
+    }
+
+    @Test
+    public void testRemotePartitionDeleteMetadataSerde() {
+        // Create RemoteLogMetadataContext for RemotePartitionDeleteMetadata
+        RemotePartitionDeleteMetadata remotePartitionDeleteMetadata = createRemotePartitionDeleteMetadata();
+
+        doTestRemoteLogMetadataSerde(remotePartitionDeleteMetadata);
+    }
+
+    private RemoteLogSegmentMetadata createRemoteLogSegmentMetadata() {
+        Map<Integer, Long> segLeaderEpochs = new HashMap<>();
+        segLeaderEpochs.put(0, 0L);
+        segLeaderEpochs.put(1, 20L);
+        segLeaderEpochs.put(2, 80L);
+        RemoteLogSegmentId remoteLogSegmentId = new RemoteLogSegmentId(TP0, Uuid.randomUuid());
+        return new RemoteLogSegmentMetadata(remoteLogSegmentId, 0L, 100L, -1L, 1,
+                                            time.milliseconds(), 1024, segLeaderEpochs);
+    }
+
+    private RemoteLogSegmentMetadataUpdate createRemoteLogSegmentMetadataUpdate() {
+        RemoteLogSegmentId remoteLogSegmentId = new RemoteLogSegmentId(TP0, Uuid.randomUuid());
+        return new RemoteLogSegmentMetadataUpdate(remoteLogSegmentId, time.milliseconds(),
+                                                  RemoteLogSegmentState.COPY_SEGMENT_FINISHED, 2);
+    }
+
+    private RemotePartitionDeleteMetadata createRemotePartitionDeleteMetadata() {
+        return new RemotePartitionDeleteMetadata(TP0, RemotePartitionDeleteState.DELETE_PARTITION_MARKED,
+                                                 time.milliseconds(), 0);
+    }
+
+    private void doTestRemoteLogMetadataSerde(RemoteLogMetadata remoteLogMetadata) {
+        RemoteLogMetadataSerde serde = new RemoteLogMetadataSerde();
+
+        // Serialize metadata and get the bytes.
+        byte[] metadataBytes = serde.serialize(remoteLogMetadata);
+
+        // Deserialize the bytes and check the RemoteLogMetadata object is as expected.
+        RemoteLogMetadataSerde deserializer = new RemoteLogMetadataSerde();
+        RemoteLogMetadata deserializedRlmContext = deserializer.deserialize(metadataBytes);
```

Comment:
Should we name deserializedRlmContext differently since there is no longer a RlmContext?